.docker/Dockerfile

```diff
@@ -1,36 +1,40 @@
+# syntax=docker/dockerfile:1.7-labs
+# Above must be the first line in Dockerfile.  It enables the "copy --parents" syntax
 ARG ROS_DISTRO=rolling
 FROM ros:$ROS_DISTRO-ros-base AS ci
 
 ENV DEBIAN_FRONTEND=noninteractive
 
-WORKDIR /root/ws_blue
-COPY . src/blue
-
 # Install apt packages needed for CI
 RUN apt-get -q update \
     && apt-get -q -y upgrade \
     && apt-get -q install --no-install-recommends -y \
-    git \
-    sudo \
     clang \
     clang-format-14 \
     clang-tidy \
     clang-tools \
-    python3-pip \
-    python3-dev \
-    lsb-release \
-    wget \
+    git \
     gnupg \
+    lsb-release \
+    python3-dev \
+    python3-pip \
     software-properties-common \
+    sudo \
+    wget \
     && apt-get autoremove -y \
     && apt-get clean -y \
     && rm -rf /var/lib/apt/lists/*
 
-# Install all ROS dependencies for _just_ blue
-# (we have not imported other repos from .repos files)
+WORKDIR /root/ws_blue/src/blue
+COPY --parents */package.xml .
+COPY blue.repos .
```

Comment:
The reason that I had this in the `robot` stage initially was to keep this image relatively lightweight for CI (the tests don't depend on the packages from `blue.repos`). I'm open to hearing your reasoning for the change though

Reply:
This change specifically is a simple replacement for the previous "COPY blue ."  It only copies the package.xml files from "blue/*" rather than all of the not-docker-ignored files in Blue to reduce the dependencies which may break cache.

Address your question below...